### DISCUSSION TITLE
[Support] Integrating Isso into a django-powered site

### DISCUSSION THREAD

**Discussion by ogurec-ogurec** [posativ/isso#1010] [Q&A] [answered]
what am I doing wrong? 

1. **pip install isso** 
2. add '_isso'_ app ot setting _INSTALLED_APPS_. 
3. Create _**isso.conf**_ with the following value: 

```
[general]
dbpath = ./comments.db 
host = 
        127.0.0.1:8000
   	http://127.0.0.1:8000/
	https://localhost:8000
	http://www.localhost:8000
	https://www.localhost:8000


[server]
listen = http://localhost:1234
[admin]
enabled = true
password = password
```
4. where is the embed-min.js file located? I am download to: 
https://github.com/redradishtech/isso/blob/master/isso/js/embed.min.js

5. Added embed.min.js to the static file
6. Added html script to template: 

```
 <script type="module" data-isso="http://127.0.0.1:8000/isso/" src="/static/isso/embed.min.js"></script>
 <section id="isso-thread"></section>
```
7. I get an error in the browser console:

```
       POST http://127.0.0.1:8000/isso/count 404 (Not Found)
       GET http://127.0.0.1:8000/isso/?uri=%2Fquestion%2Fnewpost&nested_limit=5 404 (Not Found)
       POST http://127.0.0.1:8000/isso/preview 404 (Not Found)

```

can I somehow run isso-comments on a local machine (built-in Django server - routing by url) without an HTTP server (nginx)? Or do I really need an HTTP server?

**Comment by ix5** [comment, marked as answer]
Without more information about your django setup, it will be hard to help you troubleshoot.

I personally have no experience with django, but here's what I can say so far about why what you're currently doing is definitely not working:

1. From your config, it seems like you're running django on port `8000` and expect Isso to be available from a sub-path at `localhost:8000/isso` (i.e. on the same port `8000`), yet I don't see you following the [Sub-URI approach from the docs](https://isso-comments.de/docs/reference/multi-site-sub-uri/#sub-uri)
2. You have Isso expect itself to be listening on port `1234`:
    ```
    listen = http://localhost:1234
    ```
3. You've copied the embed script from a random GitHub repo that has not been updated since 2022 (https://github.com/redradishtech/isso/blob/master/isso/js/embed.min.js)

You're not telling us how you are starting Isso (if at all) and I don't think django's auto-discovery system will automagically discover Isso as a module and run it appropriately.

---

While you can try to corral Isso into working as a django module, you will have much more success in following the Isso docs.

In short:

1. Install Isso via `pip`
2. Specify the port you want Isso to run on in `server.listen`, e.g. port `8080`, assuming you're running django on a different port, e.g. `8000`
3. Run Isso via `isso -c config.conf` with your config
4. Don't hardcode your `embed.js` as a static file but get the latest version from Isso's endpoint:
   ```
    <script data-isso="http://localhost:8080/"
            src="http://localhost:8080/js/embed.min.js"></script>
    ```
    N.b. the setting for `data-isso` pointing to `localhost:8080`

**Comment by ogurec-ogurec** [comment]
@ix5 Thank you very much, it worked!

1. Now there are two problems: when I try to edit a comment I get an error in the console:

       GET http://localhost:8080/id/13?plain=1 403 (FORBIDDEN)

2. When I try to give a rating, the rating occurs without errors in the console, but the number is not added next to the vote, and if you look in the code, there is 0. Although in the settings I set:

```
data-isso-vote="true"
data-isso-vote-levels="-1,1"
```

What can be wrong? please tell

**Reply by ix5** [reply]
The first issue _could_ be related to cookies (i.e. django is possibly set up to block third-party cookies with Isso being considered a third party). Take a look in your browser devtool console and check the cookie jar and that the correct cookie is being sent with the request (`13` or `isso-13` for your case).

The second issue is due to Isso's `voter` field being initialized to the commenter's IP address so that they cannot up-/downvote their own comment, see:
https://github.com/isso-comments/isso/blob/c257a4fe9490632bc29abc1dc294847bab06fcc7/isso/db/comments.py#L90

That IP address is `127.0.0.1` since you are the client running on `localhost`. It's not ideal for testing but it'll work once you're on a real server with real IPs.

**Reply by ogurec-ogurec** [reply]
@ix5  If I did everything correctly, this is the cookie that is sent:
![image](https://github.com/isso-comments/isso/assets/66796268/253dd0a0-1405-41d1-be42-e4b0d7bcabf3)

**Reply by ix5** [reply]
Yup, `ckCsrfToken` should be from `CKEditor` and the regular `csrftoken` from django... your Isso cookies are being rejected.

As this is no longer Isso-specific advice, I would recommend you familiarize yourself with basic web security technologies and try figuring out why some cookies might be rejected based on their origin.

Without knowing your setup, I will stop monitoring this thread because there is too much guesswork with too many components involved. You will have to find someone to help you debug that is familiar with your specific code, or learn how to debug these issues yourself.

**Reply by ogurec-ogurec** [reply]
@ix5 what settings need to be provided? I'll throw everything off, I just don't know what settings are needed for debugging. :(

**Reply by ix5** [reply]
The key word here is _reproduce_. I cannot reproduce your error. Either you produce a minimal _working_ example that anyone can run to show the issue you're experiencing (preferred), or you throw your whole django app into some repo, _with full instructions on how to run it_.

You'll have to accept that few people will want to play the debug game through guesswork with you.

Further still, even if you provided what was necessary, you're asking someone else to do the work for you. This is not a support forum for general web development. Please stop @-ing me.
